### DISCUSSION TITLE
Active link class for urlFor and linkTo?

### DISCUSSION THREAD

**Discussion by MvdO79** [wheels-dev/wheels#1420] [Ideas]
Might be an idea to have the possibility to set an active link class in case the urlFor or linkTo is currently active?

**Comment by chapmandu** [comment]
You can use a class attribute on linkTo with any value you choose. Any attribute outside those documented are rendered inline. 

Wheels has no idea what "active" means so you would need to implement the logic for the value yourself. 

I don't see a class attribute being relevant on urlFor though.

**Comment by MvdO79** [comment]
Your right  urlFor isn't relevant.

For linkTo I thought about an option like "showWhenActive=1" would be handy.
Then some logic would check if the linkTo is currently active (checked with cgi values or build in wheels params).
When active this could create an additional "active" to the css field.

Right now I find myself creating a lot of if .... then "active" in the links which is fine but could be handled better imo.

**Comment by chapmandu** [comment]
One way to tackle this is creating an application plugin where you can override the wheels linkTo function. You can implement your class logic then return the core.linkTo().

**Reply by MvdO79** [reply]
Nice, thanks for sharing.
